README.md

```diff
@@ -2,31 +2,322 @@
+
+
+Now we can evolve the `main` method to use the `Shape` interface
+as the declared type for the array.  We can add to the array
+ `Rectangle` and `Circle` objects since both classes implement
+the `Shape` interface.
+
+```java
+Shape[] tables = new Shape[] {
+        new Rectangle(7, 5),
+        new Rectangle(2, 4),
+        new Rectangle(3, 5),
+        new Circle(4),
+        new Circle(6)
+};
+```
+
+We also need to change the declared type of the loop variable to `Shape`:
+
+```java
+for (Shape table: tables) {
+    totalArea += table.getArea();
+    totalCircumference += table.getCircumference();
+}
+```
+
+The updated `Main` class is shown below:
+
+```java
+public class Main {
+    public static void main(String[] args) {
+        Shape[] tables = new Shape[] {
+                new Rectangle(7, 5),
+                new Rectangle(2, 4),
+                new Rectangle(3, 5),
+                new Circle(4),
+                new Circle(6)
+        };
+
+        double totalArea = 0;
+        double totalCircumference = 0;
+        for (Shape table: tables) {
+            totalArea += table.getArea();
+            totalCircumference += table.getCircumference();
+        }
+
+        System.out.println("Total area = " + totalArea);
+        System.out.println("Total circumference = " + totalCircumference);
+    }
+}
+```
+
+The program now includes the area and circumference of the
+two `Circle` objects in calculating the totals:
+
+```text
+Total area = 221.36281798666926
+Total circumference = 114.83185307179586
+```
+
+## Debugging the application
+
+Let's use the debugger to step through the code and understand how the `Shape` array is used.
+
+Set a breakpoint at the loop, after the array has been filled with the 3 rectangles
+and 2 circles.  Start the debug session and switch to the Java Visualizer view:
+
+![interface set breakpoint](https://curriculum-content.s3.amazonaws.com/6677/pillars/interface_breakpoint.png)
+
+Press "Step Into" to execute the first iteration of the loop, which will assign the
+loop variable `table` to point at the first element in the array.
+
+![loop step 1](https://curriculum-content.s3.amazonaws.com/6677/pillars/interface_step1.png)
+
+Press "Step Into" to execute the method call `table.getArea()`, transferring control **into**
+the method.  Since `table` references an instance of `Rectangle`, the `getArea()` method
+defined in the `Rectangle` class is called.
+
+![loop step 2](https://curriculum-content.s3.amazonaws.com/6677/pillars/interface_step2.png)
+
+Press "Step Into" to calculate and return the area to the `main` method.
+
+Keep pressing "Step Into" to loop through the rest of the elements in the array, calling `getArea()` and
+`getCircumference()` on each shape.  Confirm when the shape is a `Circle` object,
+the `getArea()` in `Circle` is executed:
+
+![loop step 3](https://curriculum-content.s3.amazonaws.com/6677/pillars/interface_step3.png)
+
+## Polymorphism
+
```

Comment:
Consider adding the sentence: "What we just saw above is an example of polymorphism." 

Better lead in maybe? Just a consideration.

Reply:
yes, thanks